client/src/pages/makerspace_page/equipment_pages/EquipmentInformation.tsx

```diff
@@ -88,6 +88,24 @@ export default function EquipmentInformation(props: EquipmentInformationProps) {
     }
   }, [moduleIDs]);
 
+  window.addEventListener('beforeunload', (event) => {
+    const unsaved = name !== props.equipment.name ||
+      imageUrl !== props.equipment.imageUrl ||
+      sopUrl !== props.equipment.sopUrl ||
+      notes !== props.equipment.notes ||
+      byReservation !== props.equipment.byReservationOnly ||
+      needsWelcome !== props.equipment.needsWelcome ||
+      requiresTrainer !== props.equipment.requiresTrainerApproval ||
+      requiresInPerson !== props.equipment.requiresInPerson ||
+      room.id !== props.equipment.room.id ||
+      moduleIDs.length !== props.equipment.trainingModules.length;
+
+    if (unsaved) {
+      event.preventDefault();
+      return '';
+    }
+  });
+
```

Comment:
not a you issue but i just checked on line 153 and you can delete the comment block and ts-ignore now, seeminlgy the api has changed since it was written